packages/manager/apps/ips/src/data/hooks/catalog/useGetCatalog.ts

```diff
@@ -0,0 +1,137 @@
+import { useQuery } from '@tanstack/react-query';
+import { apiClient } from '@ovh-ux/manager-core-api';
+import { ShellContext } from '@ovh-ux/manager-react-shell-client';
+import { useContext } from 'react';
+
+export const BYOIP_FAILOVER_V4 = 'byoip-failover-v4';
+export const CONFIG_NAME = {
+  CAMPUS: 'campus',
+  IPRIR: 'ipRir',
+} as const;
+
+export type Campus = {
+  name: string;
+  planCode: string;
+};
+
+export type ProductConfiguration = {
+  name: string;
+  values: string[] | Campus[];
+};
+
+export type Product = {
+  description: string;
+  internalType: string;
+  name: string;
+  configurations: ProductConfiguration[];
+};
+
+export type PlanDetails = {
+  metadatas: Record<string, unknown>;
+  pricings: Record<string, unknown>;
+  product: Product;
+};
+
+export type Plan = {
+  addonsFamily: string;
+  consumptionBillingStrategy: string;
+  details: PlanDetails;
+  invoiceName: string;
+  pricingType: string;
+  planCode: string;
+};
+
+export type CatalogResponse = {
+  plans: Plan[];
+};
+
+export const useGetCatalog = () => {
+  const { shell } = useContext(ShellContext);
+  const { environment } = shell;
+
+  return useQuery({
+    queryKey: ['catalog', 'bringYourOwnIp'],
+    queryFn: async () => {
+      const env = await environment.getEnvironment();
+      const { ovhSubsidiary } = env.getUser();
+
+      const { data } = await apiClient.v6.get<CatalogResponse>(
+        '/order/catalog/formatted/bringYourOwnIp',
+        {
+          params: {
+            ovhSubsidiary,
+          },
+        },
+      );
+
+      const campusList: Campus[] = [];
+      let plan: Plan | null = null;
+
+      data.plans.map((el) => {
```

Comment:
```suggestion
      data.plans.forEach((el) => {
```
Since you don't need the resulting array